test/integration/controller_test.go

```diff
@@ -423,9 +474,18 @@ func TestCAPIClusterCreate(t *testing.T) {
 			}
 
 			machineSets := make([]*capiv1alpha1.MachineSet, len(tc.machineSets))
-			var masterMachinSet *capiv1alpha1.MachineSet
+			var masterMachineSet *capiv1alpha1.MachineSet
+
+			for i, clustopClusterMS := range tc.machineSets {
+
+				role := capicommon.NodeRole
+				if clustopClusterMS.NodeType == clustopv1alpha1.NodeTypeMaster {
+					role = capicommon.MasterRole
```

Comment:
Set `masterMachineSet` here rather than looking for it by looping over the roles later.

Reply:
This doesn't quite work as we don't have the capi.MachineSet here, but we need the role to make the capi.MachineSet, I will just do a similar if below instead of iterating the roles.